WordPress/Classes/Models/BasePost.m

```diff
@@ -219,4 +219,51 @@ - (NSString *)statusForDisplay
     return statusText;
 }
 
+#pragma mark - Status
+
+- (BOOL)isDraft
+{
+    return [self.status isEqualToString:BasePostStatusDraft];
+}
+
+- (BOOL)isPending
+{
+    return [self.status isEqualToString:BasePostStatusPending];
+}
+
+- (BOOL)isPublished
+{
+    return ([self.status isEqualToString:BasePostStatusPublish]
+            && [self hasRemote]);
+}
+
+- (BOOL)isPrivate
+{
+    return [self.status isEqualToString:BasePostStatusPrivate];
+}
+
+- (BOOL)isScheduled
+{
+    return ([self.status isEqualToString:BasePostStatusPublish]
+            && ![self hasRemote]);
```

Comment:
I think the date comparison might have been a better approach, until a `future` status is respected.